src/remote.ts

```diff
@@ -440,6 +441,37 @@ export class Remote {
   // updateSSHConfig updates the SSH configuration with a wildcard that handles
   // all Coder entries.
   private async updateSSHConfig() {
+    let deploymentConfig: SSHConfigResponse = defaultSSHConfigResponse
+    try {
+      deploymentConfig = await getDeploymentSSHConfig()
+    } catch (error) {
+      if (!axios.isAxiosError(error)) {
+        throw error
+      }
+      switch (error.response?.status) {
+        case 404: {
+          // Deployment does not support overriding ssh config yet. Likely an
+          // older version, just use the default.
+          deploymentConfig = defaultSSHConfigResponse
+          break
+        }
+        case 401: {
+          const result = await this.vscodeProposed.window.showInformationMessage(
+            "Your session expired...",
+            {
+              useCustom: true,
+              modal: true,
+              detail: "You must login again to access your workspace.",
+            },
+            "Login",
+          )
+          throw error
+        }
```

Comment:
What is the best way to handle this?

Reply:
This just throws some error if there is an unauthorized response code. It shouldn't happen because we are logged in, so it's an edge case that we don't need to handle all too well?